### PR TITLE
Release process

### PR DESCRIPTION
**Pull request by agarciamontoro**
- [x] Refactor release process
- [x] Release version 9.0 with integrity checks
- [x] Release version 10.0 with integrity checks
- [x] Minimal tests for version existence

I'm leaving the tests for the code for a future PR. I want to release this asap to have some coherence between `db.rnacentral`, `db.rna16s` and `db.fragments16s`.

### CODE REVIEW

File: deps.sbt [hunk 1]
```diff
@@ -1,11 +1,12 @@
 libraryDependencies ++= mainDependencies ++ testDependencies
 
 val mainDependencies = Seq(
-  "ohnosequences" %% "aws-scala-tools" % "0.20.0"
+  "ohnosequences" %% "db-rnacentral" % "0.13.1",
+  "ohnosequences" %% "s3"            % "0.2.1",
```
Comment (ncordon) [inline deps.sbt]: This line can be removed. Already included in `db-rnacentral`

File: deps.sbt [hunk 2]
```diff
@@ -1,11 +1,12 @@
...
+  "ohnosequences" %% "files"         % "0.5.0"
 )
 
 val testDependencies = Seq(
-  "ohnosequences" %% "db-rnacentral" % "0.12.2",
-  "ohnosequences" %% "fastarious"    % "0.12.0",
-  "org.scalatest" %% "scalatest"     % "3.0.4"
+  "ohnosequences" %% "fastarious" % "0.12.0",
```
Comment (ncordon) [inline deps.sbt:8]: Just a future note: `fastarious` should be removed in the future and changed to `faster`

File: deps.sbt [hunk 3]
```diff
@@ -1,11 +1,12 @@
 libraryDependencies ++= mainDependencies ++ testDependencies
 
 val mainDependencies = Seq(
-  "ohnosequences" %% "aws-scala-tools" % "0.20.0"
+  "ohnosequences" %% "db-rnacentral" % "0.13.1",
+  "ohnosequences" %% "s3"            % "0.2.1",
+  "ohnosequences" %% "files"         % "0.5.0"
```
Comment (ncordon) [inline deps.sbt]: Same

File: src/main/scala/data.scala [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+  }
+
+  /**
+    * Generator of S3 objects in a directory parametrized by a version.
+    *
+    * @param version is the version of the rna16S data whose paths we want to
+    * obtain.
+    *
+    * @return a function `String => S3Object` that, given the name of a file,
+    * return an S3 object in a fixed S3 directory, which is parametrized by the
```
Comment (ncordon) [inline src/main/scala/data.scala]: Typo:
```suggestion
    * returns an S3 object in a fixed S3 directory, which is parametrized by the
```

File: src/main/scala/data.scala [hunk 5]
```diff
@@ -0,0 +1,78 @@
+package ohnosequences.db.rna16s
+
+import ohnosequences.db.rnacentral
+import ohnosequences.s3._
+import ohnosequences.files.digest.DigestFunction
+import ohnosequences.files.File
+
+sealed abstract class Version(val name: String) {
+  val compatibleInputVersions: Set[RNACentralVersion]
```
Comment (ncordon) [inline src/main/scala/data.scala]: What does it mean to be a compatible version?

File: src/main/scala/release.scala [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+    * ID Mappings and the sequences are downloaded, from which the 16S
+    * sequences are filtered using [[rna16sIdentification.is16s]]. The result
+    * of the filter is uploaded to the object returned by [[data.sequences]]
+    * applied over the version parameter.
+    *
+    * @note This method does not check if an overwrite will happen. Use
+    * [[generateNewDB]] for that use case.
+    *
+    * @return an Error + S3Object, with a Right(s3Obj) with the S3 path of the
+    * generated fastaa if everything worked as expected or with a Left(error)
```
Comment (ncordon) [inline src/main/scala/release.scala]:
```suggestion
    * generated fasta if everything worked as expected or with a Left(error)
```

File: src/main/scala/release.scala [hunk 7]
```diff
@@ -0,0 +1,116 @@
...
+    val fastaFile              = data.local.fastaFile(localFolder)
+    lazy val rnaCentralEntries = input.rnaCentralEntries(localFolder)
+    val outputFile             = output.sequences(localFolder)
+
+    for {
+      _ <- directory.createDirectory(localFolder).left.map(Error.FileError)
+      _ <- getCheckedFileIfDifferent(inputFasta, fastaFile)
+      _ <- getCheckedFileIfDifferent(inputIdMapping, mappingsFile)
+      _ <- generateSequences(rnaCentralEntries, outputFile)
+      _ <- paranoidPutFile(outputFile, s3Sequences)
```
Comment (ncordon) [inline src/main/scala/release.scala:77]: :+1: Nice

File: src/main/scala/release.scala [hunk 8]
```diff
@@ -0,0 +1,116 @@
...
+    } yield {
+      s3Sequences
+    }
+  }
+
+  /**
+    * Read the data from `db.rnacentral`, filter the 16S sequences and upload
+    * the result to S3 if and only if the upload does not override anything.
```
Comment (ncordon) [inline src/main/scala/release.scala]: Typo: override -> overwrite
```suggestion
    * the result to S3 if and only if the upload does not overwrite anything.
```

File: src/main/scala/release.scala [hunk 9]
```diff
@@ -0,0 +1,116 @@
...
+    *
+    * For the RNACentral version associated to the version parameter, both the
+    * ID Mappings and the sequences are downloaded, from which the 16S
+    * sequences are filtered using [[rna16sIdentification.is16s]]. The result
+    * of the filter is uploaded to the object returned by [[data.sequences]]
+    * applied over the version parameter, if and only if the upload does not
+    * override anything.
```
Comment (ncordon) [inline src/main/scala/release.scala]:
```suggestion
    * overwrite anything.
```

**Review by ncordon — APPROVED** [review]
Great work on this :muscle: !

File: deps.sbt (hunk 2 above)
Comment (agarciamontoro) [reply deps.sbt:8]: Issue opened: https://github.com/ohnosequences/db.rna16s/issues/95

File: src/main/scala/release.scala (hunk 8 above)
Comment (agarciamontoro) [reply src/main/scala/release.scala]: Oh, thanks, I thought they were interchangeable! This explains it with nice examples: https://english.stackexchange.com/a/88403

File: src/main/scala/data.scala (hunk 5 above)
Comment (agarciamontoro) [reply src/main/scala/data.scala]: I just tried to explain why this was there and found no real reason, so I'm removing it. Thanks for making me think a little bit more about this :3

**Comment by agarciamontoro** [comment]
I'll merge this as soon as it is released, thanks for the review, @ncordon

**Pull request merged**